Library/Homebrew/cmd/config.rb

```diff
@@ -119,8 +119,10 @@ def describe_system_ruby
     s
   end
 
-  def java_version
-    java = `java -version 2>&1`.lines.first.chomp
+  def describe_java
+    java = which "java"
+    return "N/A" if java.nil?
+    java = `#{java} -version 2>&1`.lines.first.chomp
```

Comment:
Looks like `/usr/libexec/java_home --exec java -version` is what we want; that'll print the java version if one is installed but won't try and install it if not.